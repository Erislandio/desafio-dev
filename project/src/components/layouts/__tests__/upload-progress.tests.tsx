// UploadProgress.test.tsx
import { render, screen } from "@testing-library/react";
import UploadProgress from "../upload-progress";

jest.mock("@radix-ui/react-progress", () => ({
  Progress: ({ value, className }) => (
    <div data-testid="progress-bar" data-value={value} className={className}>
      Progress Bar
    </div>
  ),
}));

describe("UploadProgress Component", () => {
  test("renders nothing when not uploading", () => {
    const { container } = render(
      <UploadProgress progress={50} uploading={false} />
    );

    expect(container.firstChild).toBeNull();
  });

  test("renders progress bar when uploading", () => {
    render(<UploadProgress progress={50} uploading={true} />);

    expect(screen.getByTestId("progress-bar")).toBeInTheDocument();
    expect(screen.getByText("Em progresso...")).toBeInTheDocument();
    expect(screen.getByText("50%")).toBeInTheDocument();
  });

  test("handles 100% progress", () => {
    render(<UploadProgress progress={100} uploading={true} />);

    expect(screen.getByText("100%")).toBeInTheDocument();
    expect(screen.getByTestId("progress-bar")).toHaveAttribute(
      "data-value",
      "100"
    );
  });

  test("applies correct class to progress bar", () => {
    render(<UploadProgress progress={50} uploading={true} />);

    expect(screen.getByTestId("progress-bar")).toHaveClass("h-2");
  });

  test("matches snapshot when uploading", () => {
    const { container } = render(
      <UploadProgress progress={50} uploading={true} />
    );

    expect(container).toMatchSnapshot();
  });
});
