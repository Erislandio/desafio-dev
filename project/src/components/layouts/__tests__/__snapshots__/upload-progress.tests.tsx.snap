// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadProgress Component matches snapshot when uploading 1`] = `
<div>
  <div
    class="space-y-2"
  >
    <div
      class="flex justify-between text-xs"
    >
      <span>
        Em progresso...
      </span>
      <span>
        50
        %
      </span>
    </div>
    <div
      class="h-2"
      data-testid="progress-bar"
      data-value="50"
    >
      Progress Bar
    </div>
  </div>
</div>
`;
